apps/desktop/src/app/sync/destinations/nuclia-cloud.ts

```diff
@@ -56,10 +57,18 @@ class NucliaCloudKBImpl implements IDestinationConnector {
   ): Observable<void> {
     if (params && params['kb'] && data.blob) {
       const blob = data.blob;
-      const kb$ = this.kb
-        ? of(this.kb)
-        : this.nuclia.db.getKnowledgeBox(localStorage.getItem(ACCOUNT_KEY) || '', params['kb']);
-      return kb$.pipe(
+      const mimetype = lookup(filename) || 'application/octet-stream';
+      return this.getAccount().pipe(
+        tap((account) => {
+          const applicableLimit = this.isMedia(mimetype)
+            ? account.limits.upload.upload_limit_max_media_file_size
+            : account.limits.upload.upload_limit_max_non_media_file_size;
+          if (blob.size > applicableLimit) {
+            console.error(`File too large. Size=${blob.size}, limit=${applicableLimit}`);
+            throw new Error(`File "${filename}" is too large.`);
```

Comment:
shouldn't we display a toast or something for showing the details of what failed (here the size limit) not only on the console (specially because I guess the console is not really accessible in electron)?